Dockerfile

```
@@ -2,19 +2,21 @@
 
 # Build stage: Install python dependencies
 # ===
-FROM ubuntu:jammy AS python-dependencies
-RUN apt-get update && apt-get install --no-install-recommends --yes python3-pip python3-setuptools python3-distutils
+FROM ubuntu:noble AS python-dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-pip python3-setuptools python3-venv build-essential git ca-certificates
 ADD requirements.txt /tmp/requirements.txt
 RUN pip3 config set global.disable-pip-version-check true
-RUN --mount=type=cache,target=/root/.cache/pip pip3 install --user --requirement /tmp/requirements.txt
+RUN python3 -m venv /venv
+ENV PATH="/venv/bin:${PATH}"
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install --requirement /tmp/requirements.txt
 
 # Build stage: Install yarn dependencies
 # ===
-FROM node:21 AS yarn-dependencies
+FROM node:22 AS yarn-dependencies
 WORKDIR /srv
 ADD package.json .
 ADD yarn.lock .
-RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install --production
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
```

Should we keep the --production flag? Otherwise yarn would also ship devDependencies in production (or is this not the case?)
```suggestion
RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install --production
```

---

This is a Docker multi-stage build. This command is in the second stage (yarn-dependencies), which sets up a base layer image that is used later in 2 other stages: build-js and build-css.

If we add the --production flag then we don't get the devDependencies and in the following stages when we run the JS build and the CSS build then we get errors regarding missing postcss and other stuff like that.

The final image is the last part of the Dockerfile (the line that starts with "FROM ubuntu:noble" at line 46). It doesn't matter all the node_modules we installed previously because in this image we don't run any yarn install. Instead we just copy the output of the build (which is the production build because we don't give the .env file to the Dockerfile context) to the image.

If we add the --production option there then I will need to move back a lot of devDependencies to dependencies (as it was before), which I believe it could have a more negative impact when running the webpack build.

---

Oooh ok I got it now, thanks!